notifications_utils/recipients.py

```diff
@@ -147,8 +150,22 @@ def allowed_to_send_to(self):
 
     @property
     def rows(self):
-        if self.rows_as_list is None:
+        return self.try_get_rows()
+
+    def try_get_rows(self):
+        if self.rows_as_list is not None:
+            return self.rows_as_list
+
+        def cache_rows():
             self.rows_as_list = list(self.get_rows())
+
+        thread = Thread(target=cache_rows, daemon=True)
+        thread.start()
+        thread.join(timeout=self.processing_timeout)
+
+        if thread.is_alive():
+            raise RuntimeError("CSV is too big to process.")
```

Comment:
I'm guessing this won't be the content shown to the user and instead we will need to put into the admin app some handling of this and showing a relevant error message written by Karl?

Reply:
If the intention is to catch this exception in another application I think it’s good practice to make a custom exception so you don’t accidentally catch something else which might be raising a `RuntimeError`. For example an invalid phone number doesn’t raise `ValueError`, it raises `InvalidPhoneError`:
https://github.com/alphagov/notifications-utils/blob/72a5f62d474823e1eda0c0ef47825f2d9deabf2f/notifications_utils/recipients.py#L447-L448

`InvalidPhoneError` is defined as:
https://github.com/alphagov/notifications-utils/blob/72a5f62d474823e1eda0c0ef47825f2d9deabf2f/notifications_utils/recipients.py#L356-L357

Reply:
@idavidmcdonald yes, that's the idea. However, it's also not a disaster if we don't have time, since the behaviour is (largely) unchanged from the 504 users get at the moment.

@quis good point, I've introduced a new exception class for this.

Reply:
How does it look now?

Reply:
Looks good to me 👍🏻